src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -803,13 +835,41 @@ void SolverInterfaceImpl::setMeshQuadWithEdges(
     vertices[1] = &mesh->vertices()[secondVertexID];
     vertices[2] = &mesh->vertices()[thirdVertexID];
     vertices[3] = &mesh->vertices()[fourthVertexID];
-    mesh::Edge *edges[4];
-    edges[0] = &mesh->createUniqueEdge(*vertices[0], *vertices[1]);
-    edges[1] = &mesh->createUniqueEdge(*vertices[1], *vertices[2]);
-    edges[2] = &mesh->createUniqueEdge(*vertices[2], *vertices[3]);
-    edges[3] = &mesh->createUniqueEdge(*vertices[3], *vertices[0]);
+    std::array<int,4>  hull;
+    hull[0] = firstVertexID;
+    hull[1] = secondVertexID;
+    hull[2] = thirdVertexID;
+    hull[3] = fourthVertexID;
 
-    mesh->createQuad(*edges[0], *edges[1], *edges[2], *edges[3]);
+    PRECICE_CHECK(utils::unique_elements(utils::make_array(vertices[0]->getCoords(),
+                                                           vertices[1]->getCoords(), vertices[2]->getCoords(), vertices[3]->getCoords() )),
+                  "The first triangles coordinates of the vertices must be unique!");
+
+    // Don't assume connectivity. Use gift wrapping algorithm to check quad quality and connectivity.
+    mesh->computeQuadConvexityFromPoints(hull);
+
+    Eigen::VectorXd distance1(_dimensions),distance2(_dimensions);
+
+    distance1 = vertices[hull[0]]->getCoords() - vertices[hull[2]]->getCoords();
+    distance2 = vertices[hull[1]]->getCoords() - vertices[hull[3]]->getCoords();
+    PRECICE_INFO("Distance 1: " << distance1.norm() << " and Distance 2: " << distance2.norm());
```

Comment:
```suggestion
    PRECICE_DEBUG("Distance 1: " << distance1.norm() << " and Distance 2: " << distance2.norm());
```

Use INFO with care. Only if the user really needs to know this.